tools/importTool/src/main/scala/AkkaFileReaderWithActor.scala

```diff
@@ -0,0 +1,138 @@
+
+import java.io.File
+
+import akka.http.scaladsl.model.HttpProtocols.`HTTP/1.0`
+import akka.http.scaladsl.model.MediaTypes.`text/plain`
+import akka.http.scaladsl.model.{HttpEntity, HttpMethods, HttpRequest, HttpResponse, StatusCodes, _}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{FileIO, Flow, Framing, RestartSource, Sink, Source}
+import akka.{Done, NotUsed, http}
+import akka.util.{ByteString, Timeout}
+import akka.pattern.ask
+import akka.actor.{Actor, ActorSystem, DeadLetter, Props, Scheduler, Status}
+import akka.http.scaladsl.{Http, model}
+import akka.stream.ActorMaterializer
+import HttpCharsets._
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.headers.{ByteRange, RawHeader}
+
+import scala.concurrent.duration._
+import scala.concurrent.duration.{Duration, FiniteDuration}
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+case class ActorInput(inputUrl: String, outputFormat:String, cluster:String)
+
+class AkkaFileReaderWithActor extends Actor {
+
+  import system.dispatcher
+  implicit val system = ActorSystem("app")
+  implicit val scheduler = system.scheduler
+  implicit val timeout = Timeout(5 seconds)
+
+  implicit val mat = ActorMaterializer()
+  val bytesAccumulatorActor = system.actorOf(Props(new BytesAccumulatorActor()), name = "offsetActor")
+
+  case class TripleWithKey(subject: String, triple: String)
+
+  def receive: Receive = {
+    case ActorInput(inputUrl, format, cluster) => {
+      println("Starting flow.....")
+      readAndImportFile(inputUrl, format, cluster)
+    }
+
+  }
+
+  def readFileFromServer(inputUrl:String) = {
+    println("Going to send get request to inputUrl=" + inputUrl)
+      val future = bytesAccumulatorActor ? LastOffset
+      future.flatMap(x => {
+        val ranges: String = "bytes=" + x + "-"
+        println("Resending get request for ranges=" + ranges)
+        val request = Get(inputUrl).addHeader(RawHeader.apply("Range", ranges))
+        Http().singleRequest(request)
+      })
+  }
+
+  def readAndImportFile(inputUrl:String, format:String, cluster:String)= {
+    val httpResponse = readFileFromServer(inputUrl)
+    httpResponse.onComplete(
+      {
+        case Success(r) => println("Get file from server successfully")
+        case Failure(e) => {
+          println("Got a failure while reading the file, ", e.getMessage)
+          system.scheduler.scheduleOnce(7000 milliseconds, self, ActorInput(inputUrl, format, cluster))
+        }})
+    val graphResult = httpResponse.map ({ response =>
+      println("Get response status=" + response.status)
+      val contentLengthHeader = response.headers.find(_.name == "Content-Range").map(_.value).getOrElse("")
```

Comment:
It's better to use `Content-Length` header

Reply:
not found in the http response